common/src/main/java/com/genexus/CommonUtil.java

```diff
@@ -2712,14 +2714,27 @@ public static <T> T[] concatArrays(T[] first, T[] second) {
 	}
 
 
+	private static Pattern pagingSelectPattern;
```

Comment:
I think it could be better to do: 
private final static Pattern PAGING_SELECT_PATTERN= Pattern.compile("GX_ICTE\\.(\\[\\w+\\]) AS \\b(\\w+)\\b");

it seems to be a constrant so, final and uppercases are used.
Then not initialize pagingSelectPattern  inside the method only if is null and use the constant.

Reply:
Agree on using upper case. Disagree on making it static final and compiling the pattern. After all, this method is only used when using SQLServer 2005-2008 so there is no need to compile the pattern every time the module is loaded.

You have the final word on this, just wanted to give a little more context.